Core/src/V2/Iam.php

```diff
@@ -0,0 +1,249 @@
+<?php
+/**
+ * Copyright 2023 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Core\V2;
+
+use Google\ApiCore\Serializer;
+use Google\Cloud\Core\ArrayTrait;
+use Google\Cloud\Core\Iam\PolicyBuilder;
+use Google\Cloud\Iam\V1\Policy;
+use Google\Cloud\Core\RequestHandler;
+use Google\Cloud\Iam\V1\GetIamPolicyRequest;
+use Google\Cloud\Iam\V1\GetPolicyOptions;
+use Google\Cloud\Iam\V1\SetIamPolicyRequest;
+use Google\Cloud\Iam\V1\TestIamPermissionsRequest;
+
+/**
+ * IAM Manager
+ *
+ * This class is not meant to be used directly. It should be accessed
+ * through other objects which support IAM.
+ *
+ * Policies can be created using the {@see Google\Cloud\Core\Iam\PolicyBuilder}
+ * to help ensure their validity.
+ *
+ * Example:
+ * ```
+ * // IAM policies are obtained via resources which implement IAM.
+ * // In this example, we'll use PubSub topics to demonstrate
+ * // how IAM policies are managed.
+ *
+ * use Google\Cloud\PubSub\PubSubClient;
+ *
+ * $pubsub = new PubSubClient();
+ * $topic = $pubsub->topic('my-new-topic');
+ *
+ * $iam = $topic->iam();
+ * ```
+ *
+ * @internal
+ */
+class Iam
+{
+    use ArrayTrait;
+
+    /**
+     * @var RequestHandler
+     */
+    private $requestHandler;
+
+    /**
+     * @var Serializer
+     */
+    private $serializer;
+
+    private $gapic;
+
+    /**
+     * @var string
+     */
+    private $resource;
+
+    /**
+     * @var array
+     */
+    private $policy;
+
+    /**
+     * @var array
+     */
+    private $options;
+
+    /**
+     * @param RequestHandler $requestHandler
+     * @param Serializer $serializer The serializer instance to encode/decode messages.
+     * @param string $resource
+     * @access private
+     */
+    public function __construct(
+        RequestHandler $requestHandler,
+        Serializer $serializer,
+        $gapic,
```

Comment:
It would be nice to typehint this - something to think about: we could have an `IamProviderInterface` in GAX which ensured that the methods being called on this GAPIC client exist. The generator could add this when it detects an IAM mixin

I've added a feature request in GAPIC generator here: https://github.com/googleapis/gapic-generator-php/issues/679

Reply:
As mentioned above, I think this is actually a string, so it should be typehinted as such